src/main/java/org/datadog/jmxfetch/AttachApiConnection.java

```diff
@@ -48,8 +52,10 @@ private String getJMXUrlForProcessRegex(String processRegex) throws com.sun.tool
 
                 return connectorAddress;
             }
+            jvms.add( vmd.displayName() );
         }
-        throw new IOException("Cannot find JVM matching regex: " + processRegex);
+
+        throw new IOException("Cannot find JVM matching regex: '" + processRegex + "'; available JVMs (for this user account): " + jvms );
```

Comment:
This log line is directly forwarded to the `dd-agent info` command, i.e.
```
    jmx
    ---
      - instance #foobar [ERROR]: "Cannot connect to instance process_regex: test Cannot find JVM matching regex: 'test'; available JVMs (for this user account): [org.datadog.jmxfetch.App --check jmx.yaml --check_period 15000 --conf_directory /etc/dd-agent/conf.d --log_level INFO --log_location /var/log/datadog/jmxfetch.log --reporter statsd:localhost:8125 --status_location /opt/datadog-agent/run/jmx_status.yaml collect]" collected 0 metrics
      - Collected 0 metrics, 0 events & 0 service checks
```

I think it's difficult to understand what it means when being agnostic to the code. What do you think about keeping the old message ? Alternatively the message could invite the user to run the `list_jvms` command to list all available JVMs.